src/compositions/TopBar/TopBar.tsx

```diff
@@ -21,50 +26,78 @@ interface TopBarProps {
 const TopBar: FC<TopBarProps> = ({
   mobileMenuIsVisible,
   showDesktopConnectButton,
+  showDesktopUserButton,
+  account,
   onConnectButtonClick,
   onMobileMenuButtonClick,
   className = '',
 }) => {
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+
   const containerClassName = classNames('top-bar', {
     'top-bar--mobile-menu-is-visible': mobileMenuIsVisible,
   }, className);
+  const truncatedAddress = truncateAddress(account || '');
+
+  const handleUserClick = () => {
+    setIsPopupOpen(!isPopupOpen);
+  };
+
+  const handleLogoutClick = () => {
```

Comment:
propogate this back to Page component and add this there:

```
const handleLogoutClick = (): void => {
    deactivate();
    // TODO: Make proper action for this.
    clearLastProvider();
  };
```

Reply:
`deactivate` comes from `const { active, account, deactivate } = useWeb3React<Web3Provider>();`

Reply:
Don't refresh, that's the easy way out. Make sure to close the popup after clearing wallet.